### PR TITLE
Add Documentation for core Particle classes

### PR DESCRIPTION
**Pull request by Draylar**
Hi! I started a write-up of some Javadocs for the core Particle classes. I would appreciate any feedback on the documentation. I also haven't used Enigma in a while, so please let me know if you spot a syntax error or typo!

I also renamed `method_18140` from `setSprite` to `setRandomSprite`. IMO it's confusing as a caller _what_ method you're looking at (because the arguments are `Sprite` vs. `SpriteProvider`), and the implementation of this method makes it clear it is aimed at selecting a random sprite from the particle's texture atlas. I'm happy to remove this and wait if it needs to be in a separate PR or discussed more first.

### CODE REVIEW

File: mappings/net/minecraft/client/particle/SpriteBillboardParticle.mapping [hunk 1]
```diff
@@ -1,8 +1,26 @@
...
 	FIELD field_17886 sprite Lnet/minecraft/class_1058;
-	METHOD method_18140 setSprite (Lnet/minecraft/class_4002;)V
+	METHOD method_18133 ()F
+		COMMENT @return the lower U coordinate of the {@link Sprite} rendered by this particle
+	METHOD method_18134 ()F
+		COMMENT @return the upper U coordinate of the {@link Sprite} rendered by this particle
+	METHOD method_18135 ()F
+		COMMENT @return the lower V coordinate of the {@link Sprite} rendered by this particle
+	METHOD method_18136 ()F
+		COMMENT @return the upper U coordinate of the {@link Sprite} rendered by this particle
```
Comment (YanisBft) [inline mappings/net/minecraft/client/particle/SpriteBillboardParticle.mapping]:
```suggestion
		COMMENT @return the upper V coordinate of the {@link Sprite} rendered by this particle
```

**Review by apple502j — CHANGES_REQUESTED** [review]
Please revert the rename. It is extremely impactful.

**Comment by enbrain** [comment]
I added impactful/refactor labels to avoid accidental merge. Can be removed later.

File: mappings/net/minecraft/client/particle/BillboardParticle.mapping [hunk 2]
```diff
@@ -1,8 +1,17 @@
...
+		COMMENT @return the lower U coordinate of the UV coordinates used to draw this particle
 	METHOD method_18134 getMaxU ()F
+		COMMENT @return the upper U coordinate of the UV coordinates used to draw this particle
 	METHOD method_18135 getMinV ()F
+		COMMENT @return the lower V coordinate of the UV coordinates used to draw this particle
 	METHOD method_18136 getMaxV ()F
+		COMMENT @return the upper V coordinate of the UV coordinates used to draw this particle
+	METHOD method_3087 (F)Lnet/minecraft/class_703;
+		COMMENT Scales the size of this particle by the given {@code scale} amount.
+			COMMENT the amount to multiply this particle's scale by
```
Comment (enbrain) [inline mappings/net/minecraft/client/particle/BillboardParticle.mapping]: Did Enigma produce this? It looks like you documented a parameter, but the toolchain doesn't support it.

File: mappings/net/minecraft/client/particle/ParticleTextureSheet.mapping [hunk 3]
```diff
@@ -1,12 +1,23 @@
...
 	FIELD field_17829 PARTICLE_SHEET_TRANSLUCENT Lnet/minecraft/class_3999;
 	FIELD field_17830 PARTICLE_SHEET_LIT Lnet/minecraft/class_3999;
 	FIELD field_17831 CUSTOM Lnet/minecraft/class_3999;
 	FIELD field_17832 NO_RENDER Lnet/minecraft/class_3999;
 	METHOD method_18130 begin (Lnet/minecraft/class_287;Lnet/minecraft/class_1060;)V
+		COMMENT Called to set up OpenGL render state for drawing particles of a given type.
 		ARG 1 builder
+			COMMENT builder the buffer particles will draw to in {@link net.minecraft.client.particle.Particle#buildGeometry(VertexConsumer, Camera, float)}
 		ARG 2 textureManager
+			COMMENT textureManager texture loading context
```
Comment (enbrain) [inline mappings/net/minecraft/client/particle/ParticleTextureSheet.mapping]:
```suggestion
		ARG 1 builder
			COMMENT the buffer particles will draw to in {@link net.minecraft.client.particle.Particle#buildGeometry(VertexConsumer, Camera, float)}
		ARG 2 textureManager
			COMMENT texture loading context
```

File: mappings/net/minecraft/client/particle/BillboardParticle.mapping (hunk 2 above)
Comment (Draylar) [reply mappings/net/minecraft/client/particle/BillboardParticle.mapping]: Oops. Maybe because it's an `@Override`? Removed it in the last commit.

Comment (apple502j) [reply mappings/net/minecraft/client/particle/BillboardParticle.mapping]: Yeah, I don't think you can currently document an overridden impl .

**Review by apple502j — APPROVED** [review]

**Review by modmuss50 — APPROVED** [review]

**Pull request merged**